gapic/src/main/com/google/gapid/views/TracerDialog.java

```diff
@@ -1411,6 +1433,10 @@ protected Control createDialogArea(Composite parent) {
     @Override
     public void create() {
       super.create();
+      Button ok = getButton(IDialogConstants.OK_ID);
+      if (ok != null) {
+        ok.setText("Close AGI");
```

Comment:
I am a bit afraid of the current "restart" requirement is not noticeable and mandatory to users enough. Technically, if a user doesn't notice, they could close this dialog by the "X" icon at the upper right/left corner and continue without restarting. If we can't control the "X" icon, maybe at least we should move the "restart required for ANGLE installation/update" string to this dialog as well.

Reply:
Actually even if they close the install dialog, the OK button of the trace dialog would still be disabled, because AGI doesn't notice the new installation until the server is restarted or the device is un- & re-plugged.

I have now added a close event handler, which always returns ok after successful installation, doesn't matter how the dialog is closed. This also works with the escape key. It also prevents the shell from being closed during APK installation, which was also possible before even though it shouldn't.

Reply:
Thanks for handling it! I think right now the flow guidance for the users is very clear!